File: app/src/main/java/com/fstyle/structure_android/data/source/local/sharedprf/SharedPrefsImpl.java
package com.fstyle.structure_android.data.source.local.sharedprf;

import android.content.Context;
import android.content.SharedPreferences;

/**
 * Created by le.quang.dao on 14/03/2017.
 */

public class SharedPrefsImpl implements SharedPrefsApi {

    private static final String PREFS_NAME = "AndroidStructureSharedPreferences";

    private SharedPreferences mSharedPreferences;

    public SharedPrefsImpl(Context context) {
        this.mSharedPreferences = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE);
    }

    @SuppressWarnings("unchecked")
    @Override
    public <T> T get(String key, Class<T> clazz) {
        if (clazz == String.class) {
            return (T) mSharedPreferences.getString(key, "");
        } else if (clazz == Boolean.class) {
            return (T) Boolean.valueOf(mSharedPreferences.getBoolean(key, false));
        } else if (clazz == Float.class) {
            return (T) Float.valueOf(mSharedPreferences.getFloat(key, 0));
        } else if (clazz == Integer.class) {
            return (T) Integer.valueOf(mSharedPreferences.getInt(key, 0));
        } else if (clazz == Long.class) {
            return (T) Long.valueOf(mSharedPreferences.getLong(key, 0));
        }
        return null;
    }

    @Override
    public <T> void put(String key, T data) {
        SharedPreferences.Editor editor = mSharedPreferences.edit();
        if (data instanceof String) {
            editor.putString(key, (String) data);
        } else if (data instanceof Boolean) {
            editor.putBoolean(key, (Boolean) data);
        } else if (data instanceof Float) {
            editor.putFloat(key, (Float) data);
        } else if (data instanceof Integer) {
            editor.putInt(key, (Integer) data);
        } else if (data instanceof Long) {
            editor.putLong(key, (Long) data);
        }
        editor.apply();
    }

    @Override
    public void clear() {
        mSharedPreferences.edit().clear().apply();
    }
}

File: app/src/main/java/com/fstyle/structure_android/data/source/local/sharedprf/SharedPrefsImpl.kt
package com.fstyle.structure_android.data.source.local.sharedprf

import android.content.Context
import android.content.SharedPreferences

/**
 * Created by le.quang.dao on 14/03/2017.
 */

class SharedPrefsImpl(context: Context) : SharedPrefsApi {

  private val mSharedPreferences: SharedPreferences

  init {
    this.mSharedPreferences = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
  }

  @Suppress("UNCHECKED_CAST")
  override fun <T> get(key: String, clazz: Class<T>): T? {
    if (clazz == String::class.java) {
      return mSharedPreferences.getString(key, "") as T
    } else if (clazz == Boolean::class.java) {
      return java.lang.Boolean.valueOf(mSharedPreferences.getBoolean(key, false)) as T
    } else if (clazz == Float::class.java) {
      return java.lang.Float.valueOf(mSharedPreferences.getFloat(key, 0f)) as T
    } else if (clazz == Int::class.java) {
      return Integer.valueOf(mSharedPreferences.getInt(key, 0)) as T
    } else if (clazz == Long::class.java) {
      return java.lang.Long.valueOf(mSharedPreferences.getLong(key, 0)) as T
    }
    return null
  }

  override fun <T> put(key: String, data: T) {
    val editor = mSharedPreferences.edit()
    if (data is String) {
      editor.putString(key, data)
    } else if (data is Boolean) {
      editor.putBoolean(key, data)
    } else if (data is Float) {
      editor.putFloat(key, data)
    } else if (data is Int) {
      editor.putInt(key, data)
    } else if (data is Long) {
      editor.putLong(key, data)
    }
    editor.apply()
  }

  override fun clear() {
    mSharedPreferences.edit().clear().apply()
  }

  companion object {

    private val PREFS_NAME = "AndroidStructureSharedPreferences"
  }
}
